recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,854 @@
+        else:
+            get(**self.conan_data["sources"][self.version],
+                    strip_root=True,
+                    destination=self.source_folder)
+
+        if self.no_copy_source:
+            self._patch_source()
+
+
+    def _third_party(self):
+        parties = {
+                # LEFT field:  target name for linking, will be used as TARGET::TARGET in package_info()
+                # RIGHT field: package/version to require
+                "cli11":             "cli11/2.2.0",
+                "double-conversion": "double-conversion/3.2.0",
+                "eigen":             "eigen/3.4.0",
+                "expat":             "expat/2.4.8",
+                "exprtk":            "exprtk/0.0.1",
+                "fmt":               "fmt/8.1.1",      # 9.1.0 release docs mention a PR - confirmed merged 8.1.0
+                "freetype":          "freetype/2.12.1",
+                "glew":              "glew/2.2.0",
+                "jsoncpp":           "jsoncpp/1.9.5",
+                "libharu":           "libharu/2.4.3",
+                "kissfft":           "kissfft/131.1.0",
+                "lz4":               "lz4/1.9.3",
+                "libpng":            "libpng/1.6.37",
+                "proj":              "proj/9.0.1", # if MAJOR version changes, update ThirdParty/libproj/CMakeLists.txt
+                "pugixml":           "pugixml/1.12.1",
+                "sqlite3":           "sqlite3/3.38.5",
+                "utfcpp":            "utfcpp/3.2.1",
+                "xz_utils":          "xz_utils/5.2.5", # note: VTK calls this lzma
+                "zlib":              "zlib/1.2.12",
+                "jpeg":              "libjpeg-turbo/2.1.2",
+                "TIFF":              "libtiff/4.3.0",
+                }
+
+        if (self.options.build_all_modules
+                or self.options.group_enable_StandAlone):
+            parties["hdf5"]    = "hdf5/1.13.1"
+            parties["theora"]  = "theora/1.1.1"
+            parties["ogg"]     = "ogg/1.3.5"
+            parties["netcdf"]  = "netcdf/4.8.1"
+            parties["libxml2"] = "libxml2/2.9.14"
+            parties["cgns"]    = "cgns/4.3.0"
+
+        # unused
+        if False:
+            parties["zfp"]     = "zfp/0.5.5"
+
+        if self.options.build_all_modules:
+            parties["boost"]  = "boost/1.79.0"
+            parties["openvr"] = "openvr/1.16.8"
+            parties["odbc"]   = "odbc/2.3.9"
+
+        if (self.options.build_all_modules
+                or self.options.group_enable_Qt
+                or self.options.module_enable_GUISupportQt
+                or self.options.module_enable_GUISupportQtQuick
+                or self.options.module_enable_GUISupportQtSQL
+                or self.options.module_enable_RenderingQt
+                or self.options.module_enable_ViewsQt):
+            parties["qt"] = "qt/6.3.1"
```

Comment:
You have a bug.
Here:
```python
 if (self.options.build_all_modules
                or self.options.group_enable_Qt
                or self.options.module_enable_GUISupportQt
                or self.options.module_enable_GUISupportQtQuick
                or self.options.module_enable_GUISupportQtSQL
                or self.options.module_enable_RenderingQt
                or self.options.module_enable_ViewsQt):
                ```
will always evaluate to true because all the Qt parameters are all non-empty strings which evaluate to a truthy value.

Reply:
Oh good point, i've pushed a commit without testing - see what you think